### PR TITLE
apiserver: add flag that disable the proxy handlers

### PR DESCRIPTION
**Pull request by mikedanese**
Fixes: https://github.com/kubernetes/kubernetes/issues/59885

```release-note
Add a flag that disables proxy handlers on resources that support proxying.
```

### CODE REVIEW

File: cmd/kube-apiserver/app/options/options.go [hunk 1]
```diff
@@ -150,6 +151,9 @@ func (s *ServerRunOptions) AddFlags(fs *pflag.FlagSet) {
 	fs.BoolVar(&s.EnableLogsHandler, "enable-logs-handler", s.EnableLogsHandler,
 		"If true, install a /logs handler for the apiserver logs.")
 
+	fs.BoolVar(&s.EnableProxyHandlers, "enable-proxy-handlers", s.EnableProxyHandlers,
+		"If true, install a /proxy handlers on resources that support proxying.")
```
Comment (cjcullen) [inline cmd/kube-apiserver/app/options/options.go]: "install a /proxy handler"

**Comment by lavalamp** [comment]
Seems harmless, although it's yet another flag that needs to be set the same on all apiservers. Do conformance tests still pass if it's turned off?

/lgtm
/approve

**Comment by lavalamp** [comment]
cc @deads2k just fyi

**Comment by mikedanese** [comment]
@lavalamp no, conformance tests do not pass with this turned off. Proxying functionality is explicitly part of conformance:

https://github.com/kubernetes/kubernetes/blob/master/test/conformance/testdata/conformance.txt#L134-L136

**Comment by deads2k** [comment]
@mikedanese @lavalamp Selectively disabling apiserver subresource structs across multiple resources seems worthy of a discussion.  Why would we treat these subresources specially?  Why wouldn't we handle these like any other sensitive endpoint and say that an authorizer with properly configured rules should protect you?

@liggitt @sttts Does this seem weird to you too?

/hold

holding for at least a brief discussion.  Wiring one off flags for disabling a few endpoints that don't appear to be riskier than others seems off.

**Comment by sttts** [comment]
> say that an authorizer with properly configured rules should protect you

this. +1

**Comment by mikedanese** [comment]
@cjcullen, @liggitt, @deads2k and I talked about this offline. I'll propose a path forward in 1.11 on the bug.

**Pull request closed without merging**